index.bs

```diff
@@ -4332,6 +4344,22 @@ as seen for example in [[#example-ws-no-backpressure]].
  1. Return ! [$IsWritableStreamLocked$]([=this=]).
 </div>
 
+<div algorithm>
+ The <dfn id="ws-write" method for="WritableStream">write(|chunk|)</dfn> method steps are:
+
+ 1. Let |result| be ? [$AcquireWritableStreamDefaultWriter$]([=this=]).
+ 1. If |result| is an abrupt completion, return [=a promise rejected with=] |result|.\[[Value]].
+ 1. Let |writer| be |result|.\[[Value]].
+ 1. Let |writePromise| be ! [$WritableStreamDefaultWriterWrite$](|writer|, |chunk|).
+ 1. Return the result of [=reacting=] to |writePromise|:
+  1. If |writePromise| is fulfilled with a value |v|, then:
+   1. Perform ! [$WritableStreamDefaultWriterRelease$](|writer|).
+   1. Return |v|.
+  1. If |writePromise| is rejected with a value |r|, then:
+   1. Perform ! [$WritableStreamDefaultWriterRelease$](|writer|).
+   1. Return [=a promise rejected with=] |r|.
```

Comment:
We don't need to wait for the `writePromise` to resolve or reject before releasing the lock. Instead, we should release it *immediately*.
```suggestion
 1. Let |writePromise| be ! [$WritableStreamDefaultWriterWrite$](|writer|, |chunk|).
 1. Perform ! [$WritableStreamDefaultWriterRelease$](|writer|).
 1. Return |writePromise|.
```
This works, because [WritableStreamDefaultWriterRelease](https://streams.spec.whatwg.org/#writable-stream-default-writer-release) does not reject any pending write requests.

This also resolves https://github.com/whatwg/streams/pull/1339#issuecomment-2611228239.